test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5296,7 +5296,7 @@ def save_configuration(self, outfile, mode='a'):
         else:
             conf['qmgr_print_sched'] = ret['out']
         ret = self.du.run_cmd(self.hostname, [pbsnodes, '-av'],
-                              logerr=False, level=logging.DEBUG)
+                              logerr=False, level=logging.DEBUG, sudo=True)
```

Comment:
why does this need a sudo ? it's just pbsnodes -av command right?

Reply:
`pbsnodes -av` without `sudo` i.e. if this command is run as a test execution user, it does not list all the nodes attributes like Port, version etc. We just proceed with saving the configuration without `Port`. While loading configuration, we create the node using `Port` and that gives a `KeyError`. So, I had to add this.

Reply:
thanks for explaining, can you please add this as a comment in the code so that someone like me doesn't remove it thinking it's unnecessary ? :)

Reply:
Added the comment. Also, I have updated the design document. Please review both.

Reply:
As a note for the future, you didn't need to use sudo.  There is no difference between running as root and another manager.  You could have done a runas=MGR_USR and it would have been a bit more secure.